### PR TITLE
Changed BodyStream parameter in upload functions from pointer to refe…

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by katmsft — APPROVED** [review]

File: sdk/storage/azure-storage-files-datalake/CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@
 - Removed `GetUserDelegationKeyResult`.
 - Removed `DataLake` from the names of return types and option types.
 - Removed `RequestId` from the return types.
+- Changed `BodyStream` parameter in upload functions from pointer to reference.
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-datalake/CHANGELOG.md:23]: There is no datalake upload function that got updated here. Only Append.

File: sdk/storage/azure-storage-blobs/CHANGELOG.md [hunk 2]
```diff
@@ -28,6 +28,7 @@
 - Removed `Blob` from the name of lease options and return types, like `AcquireBlobLeaseOptions` was renamed to `AcquireLeaseOptions`.
 - Removed `Blob` prefix from the name of structs `BlobGeoReplication`, `BlobLeaseStatus`, `BlobLeaseState`, `BlobLeaseDurationType`, `BlobAnalyticsLogging`, `BlobMetrics`, `BlobCorsRule`, `BlobRetentionPolicy`, `BlobStaticWebsite`, `BlobArchiveStatus` and `BlobGeoReplictionStatus`.
 - Removed `RequestId` from return types.
+- Changed `BodyStream` parameter in upload functions from pointer to reference.
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-blobs/CHANGELOG.md:31]: Would we consider append/stage block as "upload"? We may want to include the other APIs too in the CL.

**Comment by ahsonkhan** [comment]
What about some of these static methods in `blob_rest_client.hpp` (found 5 places). Everything else looks to be fixed:
https://github.com/Azure/azure-sdk-for-cpp/blob/1b316d3b44f8c38c3de80b2115f2050496d646d0/sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp#L6693-L6696

**Pull request merged**